### PR TITLE
Adding a progress bar

### LINKED ISSUE #558: Progress bar.
I'm playing around with a progress bar using https://pypi.python.org/pypi/tqdm, I'll throw in some gifs below of it working as a proof of concept. If we like it, I'll open a PR (my thinking is to make it optional).

### PR DESCRIPTION
**Pull request by drvinceknight**
Closes #558

This is built on top of #572.

Works for both parallel and serial. Here's a gif showing it (code from https://gist.github.com/drvinceknight/2f42efc2388f41d3da7931ffc10aee6f):

941gayzwqh

### CODE REVIEW

**Comment by marcharper** [comment]
Very nice :)

File: axelrod/tournament.py [hunk 1]
```diff
@@ -66,7 +68,7 @@ def setup_output_file(self, filename=None):
         # Save filename for loading ResultSet later
         self.filename = filename
 
-    def play(self, build_results=True, filename=None):
+    def play(self, build_results=True, filename=None, progress_bar=None):
```
Comment (meatballs) [inline axelrod/tournament.py]: If progress_bar is a boolean, why not use False as the default value?

File: axelrod/tournament.py [hunk 2]
```diff
@@ -76,20 +78,28 @@ def play(self, build_results=True, filename=None):
...
         filename : string
             name of output file
+        progress_bar : bool
+            Whether or not to create a progress bar which will be updated
 
         Returns
         -------
         axelrod.ResultSet
         """
+        if progress_bar is not None:
```
Comment (meatballs) [inline axelrod/tournament.py]: and this could become ```if not progress_bar```

File: axelrod/tournament.py [hunk 3]
```diff
@@ -107,13 +117,25 @@ def _build_result_set(self):
...
+        progress_bar : bool
+            Whether or not to update the tournament progress bar
+        """
         chunks = self.match_generator.build_match_chunks()
 
         for chunk in chunks:
             results = self._play_matches(chunk)
             self._write_interactions(results)
+
+            if progress_bar is not None:
```
Comment (meatballs) [inline axelrod/tournament.py]: same comment as previous

File: axelrod/tournament.py [hunk 4]
```diff
@@ -202,6 +234,9 @@ def _process_done_queue(self, workers, done_queue):
                 stops += 1
             else:
                 self._write_interactions(results)
+
+                if progress_bar is not None:
```
Comment (meatballs) [inline axelrod/tournament.py]: same again

**Comment by drvinceknight** [comment]
Good call @meatballs, have made that change (and also added some more tests that check if the progress bar was or wasn't created).

**Comment by drvinceknight** [comment]
Have added a line to the docs about the progress bar @meatballs: 08a0bb5

I don't feel strongly about it. Just a thought. Happy to throw it out...

**Comment by meatballs** [comment]
:+1:

Vince Knight wrote:

**Comment by marcharper** [comment]
It looks good for me, my only comment is that we might want progress bars on by default. Regardless this is :+1: from me.

**Comment by drvinceknight** [comment]
I'm easy.

Note that with False by default when running the test suite you get 2 or 3
progress bars appear in the output. Changing it to True by default would
have a BUNCH more of them. Not the end of the world.

Any preference @meatballs?

**Comment by meatballs** [comment]
How about True by default but the test suite passes in False?

**Comment by drvinceknight** [comment]
That would work. I'll do that now on the train. :)

**Comment by drvinceknight** [comment]
> How about True by default but the test suite passes in False?

That's done. A few progress bar still appear (to test that it actually works and doesn't break anything) but most are turned off.

**Pull request merged**